docs/training-apis/api-pytorch-advanced.txt

```diff
@@ -62,16 +62,11 @@ training") is easy if you follow a few rules.
    wrap your sampler or batch_sampler. For your training dataset, make sure that you always repeat
    AFTER you shuffle, otherwise your shuffle will hang.
 
--  **Always shard, and not before a repeat**: Use Determined's DistributedSampler or
-   DistributedBatchSampler to provide a unique shard of data to each worker based on your sampler or
-   batch_sampler. It is best to always shard your data, and even when you are not doing distributed
-   training, because in non-distributed-training settings, the sharding is nearly zero-cost, and it
-   makes distributed training seamless if you ever want to use it in the future.
-
-   It is generally important to shard after you repeat, unless you can guarantee that each shard of
-   the dataset will have the same length. Otherwise, differences between the epoch boundaries for
-   each worker can grow over time, especially on small datasets. If you shard after you repeat, you
-   can change the number of workers arbitrarily without issue.
```

Comment:
This is actually still true... just the on_epoch_ends will do a better job at pretending it's not true.

Reply:
Yeah, my thinking here was that if all of the data is shuffled anyway, then the sharding/repeating order hardly matters, but this makes too many assumption. I'll revert it.